### DISCUSSION TITLE
TRPCClientError : Cannot read properties of undefined (reading 'message')

### DISCUSSION THREAD

**Discussion by hasithaweerasinghe** [trpc/trpc#4777] [Q&A]
Hi, I'm getting following error when I run JEST tests with [msw-trpc](https://github.com/maloguertin/msw-trpc). This error occurred  after updating tRPC. 

Error after updating these package versions : 
```
"@trpc/client": "^10.8.1"       => "@trpc/client": "^10.34.0",
"@trpc/next": "^10.8.1"         => "@trpc/next": "^10.34.0",
"@trpc/react-query": "^10.8.1"  => "@trpc/react-query": "^10.34.0",
"@trpc/server": "^10.8.1"       => "@trpc/server": "^10.34.0",
```

Error :
```
/node_modules/@trpc/client/dist/TRPCClientError-dffb44e0.js:12
            return new TRPCClientError(cause.error.message ?? '', {
                                                   ^
TypeError: Cannot read properties of undefined (reading 'message')

    at Function.from (/node_modules/@trpc/client/dist/TRPCClientError-dffb44e0.js:12:52)
    at /node_modules/@trpc/client/dist/httpUtils-fd6f3ad7.js:127:52
    at runNextTicks (node:internal/process/task_queues:61:5)
    at processImmediate (node:internal/timers:437:9)
 ELIFECYCLE  Test failed. See above for more details.
```

This is my mock tRPC setup:

```
import React, { ReactNode, useState } from "react";
import {
  QueryCache,
  QueryClient,
  QueryClientProvider,
} from "@tanstack/react-query";
import {
  createTRPCReact,
  getFetch,
  httpLink,
  loggerLink,
} from "@trpc/react-query";
import superjson from "superjson";
import { AppRouter } from "../../src/server/api/root";

export const mockedTrpcReact = createTRPCReact<AppRouter>({
  overrides: {
    useMutation: {
      async onSuccess(opts) {
        await opts.originalFn();
        await opts.queryClient.invalidateQueries();
      },
    },
  },
});

export const TRPCWrapper = ({ children }: { children: ReactNode }) => {
  const [queryClient] = useState(
    () =>
      new QueryClient({
        defaultOptions: {
          queries: {
            retry: false,
          },
        },
        queryCache: new QueryCache({
          onError: (error) => console.log("queryCache error", { error }),
        }),
      })
  );
  const [trpcClient] = useState(() =>
    mockedTrpcReact.createClient({
      transformer: superjson,
      links: [
        httpLink({
          url: "http://localhost:3000/trpc",
          fetch: async (input, init?) => {
            // console.log("fetching", { input }, { init });
            const fetch = getFetch();

            return fetch(input, {
              ...init,
              // credentials: "include",
            });
          },
        }),
      ],
    })
  );

  return (
    <mockedTrpcReact.Provider client={trpcClient} queryClient={queryClient}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </mockedTrpcReact.Provider>
  );
};

```

**Comment by dmitryplatonov** [comment]
I'm also having this problem with trpc 10.36.0. Were you able to solve it?

**Comment by farhan-helmy** [comment]
having the same problem, damn

**Comment by anthony-dandrea** [comment]
We experienced this problem as well and were able to fix it. Can't say for sure it's the same issue for everyone else though. 

**Detection:**
It was an error that didn't happen all the time. And sometimes only in GitHub actions and not locally. Running tests with the `--detectOpenHandles` seemed to better replicate the issue. And at least point us to which test was problematic. Then once we found we could successfully run all the tests with a `skip` for the problematic test we knew we were looking in the correct spot. 

**Fix:**
This is specific to our test. But we had an issue where the test had mocked a trpc route. But the test's assertions passed before the request completed. And it seems the test teardown happened before the trpc finished causing the error. We fixed our test to add an assertion to `waitFor` the request to complete before running our assertions and finishing our test.

**Ideal changes to trpc:**
Maybe optional chaining on `cause.error.message` to see if it bypasses this issue. Or at least some error handling with a message. Would have been easier if the stack trace at least pointed us to the bad test.

**Comment by KATT** [comment]
If you think it's a bug in trpc, please open an issue
